bin/i18n/curriculum_sync_utils.rb

```diff
@@ -202,15 +202,34 @@ def self.sync_out_reorganize
         script_objects[name] = data if data.present?
       end
 
-      # Then we recursively flatten all of our hashes of objects, and write
-      # each resulting collection of strings out to a rails i18n config file.
+      # Then we recursively flatten all of our hashes of objects, to group them
+      # by type rather than by script
       result = flatten(script_objects, ScriptCrowdinSerializer, :scripts)
+
+      # Then we apply some postprocessing.
+      # We use URLs as keys for lessons in Crowdin, to make things easier for
+      # translators. For the actual translation files, though, we'd like to use
+      # more-standard keys.
+      rekeyed_lessons = result[:lessons].map do |lesson_url, lesson_data|
+        route_params = Rails.application.routes.recognize_path(lesson_url)
+        lesson = Lesson.joins(:script).
+          find_by(
+            "scripts.name": route_params[:script_id],
+            relative_position: route_params[:position].to_i
```

Comment:
I don't think this is a safe way to look up lessons. It is possible for multiple lessons in a script to have the same `relative_position` because of lockable lessons. For example, locally I get:
```
[development] dashboard > Lesson.joins(:script).where("scripts.name": 'csd1-2021', relative_position: 1).count
   (3.1ms)  SELECT COUNT(*) FROM `stages` INNER JOIN `scripts` ON `scripts`.`id` = `stages`.`script_id` WHERE `stages`.`relative_position` = 1 AND `scripts`.`name` = 'csd1-2021'
=> 2
```

I think just querying for lessons that have `has_lesson_plan` set to true will fix this

Reply:
Ha, I actually went and looked up https://github.com/code-dot-org/code-dot-org/blob/653b730a8505385ec1085a5d4cd6d085c8b4348f/dashboard/app/controllers/lessons_controller.rb#L24-L26 to see what the right way to do this was; I noticed that it checked `has_lesson_plan` and made a mental note to figure out _why_ so I could know whether or not it needed to be included here, and then I just didn't. Thanks for catching!